src/cnd/cnd.ts

```diff
@@ -44,6 +44,50 @@ export interface SwapRequest {
   peer: Peer;
 }
 
+export interface CoreRequestBody {
+  // to_public_key: string; // This is exchanged via libp2p
+  // secret_hash: string; // Generated by cnd
+  // alpha_ledger: string; // Ledger name is in the route
+  // beta_ledger: string; // Ledger name is in the route
+  alpha_amount: string; // Asset name is in the route
+  beta_amount: string; // Asset name is in the route
+  // final_cltv_delta: number; // beta_expiry
+  // cltv_expiry: number; // beta_expiry
+  alpha_expiry: number; // Fits in JS numbers
+  beta_expiry: number; // Fits in JS numbers
+  // chain: string; // (from lnd action payload) In the URL
+  alpha_network: string; // For Ethereum we use "chain" terminology but we could revert to "network" on this new route to keep same terminology across all ledgers
+  beta_network: string; // The bitcoin network
+  // Identities are always mandatory even if cnd could discover it (for lnd).
+  // So that the interface is simpler (no optional fields) and also adds an element of security (ensure it's connected to the right lnd).
+  // Also we dropped the `redeem` and `refund` terminology as they are redundant with the `alpha/beta` qualifier.
+  // Finally, this interface is to be used by both Bob and Alice so using the `refund`/`redeem` terminology would just make it more complex
+  alpha_ledger_identity: string;
+  beta_ledger_identity: string;
+  // The same endpoint is to be used on both nodes, depending on the role the communication and protocol execution changes.
+  // `role` may not be specific enough to be clear. We could consider adding `protocol_` or `cryptographic_` prefix
+  role: "Alice" | "Bob";
+  peer: Peer; // cnd details of the counterpart
+}
+
+export interface AlphaHerc20AdditionalRequestBody {
+  // ERC20 asset need one more information not specified in the Route.
+  // Instead of making it optional for all route it should be mandatory
+  // for the route that uses it.
+  alpha_contract_address: string;
+}
+
+export interface BetaHerc20AdditionalRequestBody {
+  beta_contract_address: string;
+}
+
+export type HanEthereumEtherHalightLightningBitcoinRequestBody = CoreRequestBody;
```

Comment:
:question: Isn't it more like this?

```suggestion
export type HanEthereumEtherHalightRequestBody = CoreRequestBody;
```

The protocol implies the ledger and the asset in the case of HALight.

Reply:
We could support Litecoin over Lightning
Can we support raiden over Halight?

Reply:
I guess being specific can't hurt.